src/nfts/utils.ts

```diff
@@ -4,12 +4,11 @@ export function capitalize(str: string | undefined) {
   return str ? str.charAt(0).toUpperCase() + str.slice(1) : str;
 }
 
-export function reduceDataEntries(entries: DataTransactionEntry[]) {
-  return entries.reduce<Record<string, DataTransactionEntry['value']>>(
-    (data, item) => {
-      data[item.key] = item.value;
-      return data;
-    },
-    {}
-  );
+export function reduceDataEntries<
+  T extends { key: DataTransactionEntry['key']; value: unknown }
```

Comment:
Actually, the `entries` type is `DataTransactionEntry[]`.
Why are we omitting `type`?

Reply:
`DataTransactionEntry` is a union type that combines `DataTransactionEntryInteger`, `DataTransactionEntryString` and so on. if we pass data of type such as `DataTransactionEntryInteger` to this function, then at the output we will get a type inferred from `DataTransactionEntry`, but we expected it to be processed as `DataTransactionEntryInteger`

Reply:
Just `T extends DataTransactionEntry` would work as well